File: 统计工具/affiliate-main/affiliate-main/test-sign.js
// 测试sign函数是否正确
const crypto = require('crypto');

function generateSign(data) {
  const salt = 'TSf03xGHykY';
  const combined = data + salt;
  const hash = crypto.createHash('md5').update(combined, 'utf-8').digest('hex');
  return hash;
}

// 测试用例
console.log('🧪 Sign函数测试\n');
console.log('=' .repeat(60));

// 测试1: 简单字符串
const test1 = 'hello';
const sign1 = generateSign(test1);
console.log('\n测试1: 简单字符串');
console.log(`输入: ${test1}`);
console.log(`输出: ${sign1}`);
console.log('Python对比: 运行 python -c "import hashlib; print(hashlib.md5(\'helloTSf03xGHykY\'.encode()).hexdigest())"');

// 测试2: 登录参数
const username = 'omnilearn';
const password = 'Ltt.104226';
const code = '1234';
const remember = '1';
const timestamp = '1234567890';

const loginData = username + password + code + remember + timestamp;
const sign2 = generateSign(loginData);

console.log('\n测试2: 登录参数');
console.log(`输入数据: ${loginData}`);
console.log(`计算的sign: ${sign2}`);

// 测试3: 报表查询参数
const startDate = '2024-12-01';
const endDate = '2024-12-31';
const page = '1';
const pageSize = '2000';
const exportFlag = '0';

const reportData = `m_id${startDate}${endDate}${page}${pageSize}${exportFlag}`;
const sign3 = generateSign(reportData);

console.log('\n测试3: 报表查询参数');
console.log(`输入数据: ${reportData}`);
console.log(`计算的sign: ${sign3}`);

console.log('\n' + '='.repeat(60));
console.log('✅ Sign函数测试完成');
console.log('\n💡 如果有Python环境，可以运行以下命令对比结果:');
console.log(`   python -c "import hashlib; print(hashlib.md5('${loginData}TSf03xGHykY'.encode()).hexdigest())"`);

File: 统计工具/affiliate-main/affiliate-main/test-sign.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Sign函数Python版本 - 用于对比测试
"""

import hashlib

def sign(data, salt="TSf03xGHykY"):
    """计算签名"""
    data = data + salt
    return hashlib.md5(data.encode('utf-8')).hexdigest()

if __name__ == '__main__':
    print('🧪 Sign函数测试 (Python版本)\n')
    print('=' * 60)

    # 测试1: 简单字符串
    test1 = 'hello'
    sign1 = sign(test1)
    print(f'\n测试1: 简单字符串')
    print(f'输入: {test1}')
    print(f'输出: {sign1}')

    # 测试2: 登录参数
    username = 'omnilearn'
    password = 'Ltt.104226'
    code = '1234'
    remember = '1'
    timestamp = '1234567890'

    login_data = username + password + code + remember + timestamp
    sign2 = sign(login_data)

    print(f'\n测试2: 登录参数')
    print(f'输入数据: {login_data}')
    print(f'计算的sign: {sign2}')

    # 测试3: 报表查询参数
    start_date = '2024-12-01'
    end_date = '2024-12-31'
    page = '1'
    page_size = '2000'
    export_flag = '0'

    report_data = f'm_id{start_date}{end_date}{page}{page_size}{export_flag}'
    sign3 = sign(report_data)

    print(f'\n测试3: 报表查询参数')
    print(f'输入数据: {report_data}')
    print(f'计算的sign: {sign3}')

    print('\n' + '=' * 60)
    print('✅ Sign函数测试完成')
